### PR TITLE
Add `MeasureEachZ` operation

### PR DESCRIPTION
**Pull request by swernli**
This change adds a new `MeasureEachZ` operation that is compatible with QIR generation. The new operation is meant as a replacement to `MultiM` and that operation can be updated to call this one as part of an eventual deprecation strategy.

Resolves https://github.com/microsoft/qsharp-runtime/issues/1005

### CODE REVIEW

**Comment by swernli** [comment]
@msoeken I’ll need to get an API review done for this one to make sure I have the right approval to add this to the language. Can you help me get that started?

**Comment by msoeken** [comment]
@swernli I will add this to the May API review. Could you please create an issue related to this PR in qsharp-runtime to which I can refer to in the API review?

**Comment by swernli** [comment]
@msoeken Thanks, here it is: https://github.com/microsoft/qsharp-runtime/issues/1005

File: src/Simulation/TargetDefinitions/Decompositions/MeasureEachZ.qs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    /// ## targets
+    /// An array of qubits to be measured.
+    /// # Output
+    /// An array of measurement results.
+    ///
+    /// # Remarks
+    /// This operation does not reset the measured qubits to the |0⟩ state, 
+    /// leaving them in the state that corresponds to the measurement results.
+    @RequiresCapability(
+        "BasicQuantumFunctionality",
```
Comment (bettinaheim) [inline src/Simulation/TargetDefinitions/Decompositions/MeasureEachZ.qs:20]: I think this attribute needs to be updated.

**Comment by bettinaheim** [comment]
@swernli Incorporated into https://github.com/microsoft/qsharp-runtime/pull/1019, so I am closing this here. 
Please review there.

**Pull request closed without merging**